### ISSUE TITLE
Cannot compile any code using this library, due to a binding generation issue.

### ISSUE THREAD

**Issue by dexterdy** [StefanSalewski/gintro#214]
No code I write using this library compiles. Suspected cause is a bug in the local binding generation. I tested both the gtk3 and gtk4 app0 examples, both with the same error. 

I am using nimble to compile and run and I also installed gintro via nimble without any errors. I added gintro to my .nimble file as a dependency. I did not change the code whatsoever. I am using version 1.6.10 of the Nim compiler. I am on fedora 37. The following is the output of `nimble run --verbose` on the gtk4 example:

```
[dexterdy@fedora relmnotes]$ nimble run --verbose
  Verifying dependencies for relmnotes@0.1.0
    Reading official package list
   Checking for gintro@any version
      Info: Dependency on gintro@any version already satisfied
  Verifying dependencies for gintro@0.9.9
   Building relmnotes/realmnotes using c backend
  Executing /home/dexterdy/.nimble/bin/nim c --colors:on --noNimblePath -d:NimblePkgVersion=0.1.0 --path:/home/dexterdy/.nimble/pkgs/gintro-0.9.9 -o:/home/dexterdy/Documents/repos/relmnotes/realmnotes /home/dexterdy/Documents/repos/relmnotes/src/realmnotes.nim
Hint: used config file '/home/dexterdy/.choosenim/toolchains/nim-1.6.10/config/nim.cfg' [Conf]
Hint: used config file '/home/dexterdy/.choosenim/toolchains/nim-1.6.10/config/config.nims' [Conf]
...............................................................................................
/home/dexterdy/.nimble/pkgs/gintro-0.9.9/gintro/gtk4.nim(33095, 82) Error: type mismatch: got 'set[empty]' for '{}' but expected 'ApplicationFlags = enum'
     Error: Build failed for package: relmnotes
        ... Execution failed with exit code 1
        ... Command: /home/dexterdy/.nimble/bin/nim c --colors:on --noNimblePath -d:NimblePkgVersion=0.1.0 --path:/home/dexterdy/.nimble/pkgs/gintro-0.9.9 -o:/home/dexterdy/Documents/repos/relmnotes/realmnotes /home/dexterdy/Documents/repos/relmnotes/src/realmnotes.nim
```

The code the output refers to is from the gtk4 gintro bindings:

```
proc newApplication*(applicationId: cstring = nil; flags: gio.ApplicationFlags = {}): Application =
  let gobj = gtk_application_new(applicationId, flags)
  let qdata = g_object_get_qdata(gobj, Quark)
  if qdata != nil:
    result = cast[type(result)](qdata)
    assert(result.impl == gobj)
  else:
    fnew(result, gtk4.finalizeGObject)
    result.impl = gobj
    GC_ref(result)
    if g_object_is_floating(result.impl).int != 0:
      discard g_object_ref_sink(result.impl)
    g_object_add_toggle_ref(result.impl, toggleNotify, addr(result[]))
    g_object_unref(result.impl)
    assert(g_object_get_qdata(result.impl, Quark) == nil)
    g_object_set_qdata(result.impl, Quark, addr(result[]))

proc newApplication*(tdesc: typedesc; applicationId: cstring = nil; flags: gio.ApplicationFlags = {}): tdesc =
  assert(result is Application)
  let gobj = gtk_application_new(applicationId, flags)
  let qdata = g_object_get_qdata(gobj, Quark)
  if qdata != nil:
    result = cast[type(result)](qdata)
    assert(result.impl == gobj)
  else:
    fnew(result, gtk4.finalizeGObject)
    result.impl = gobj
    GC_ref(result)
    if g_object_is_floating(result.impl).int != 0:
      discard g_object_ref_sink(result.impl)
    g_object_add_toggle_ref(result.impl, toggleNotify, addr(result[]))
    g_object_unref(result.impl)
    assert(g_object_get_qdata(result.impl, Quark) == nil)
    g_object_set_qdata(result.impl, Quark, addr(result[]))

proc initApplication*[T](result: var T; applicationId: cstring = nil; flags: gio.ApplicationFlags = {}) {.deprecated.} =
  assert(result is Application)
  let gobj = gtk_application_new(applicationId, flags)
  let qdata = g_object_get_qdata(gobj, Quark)
  if qdata != nil:
    result = cast[type(result)](qdata)
    assert(result.impl == gobj)
  else:
    fnew(result, gtk4.finalizeGObject)
    result.impl = gobj
    GC_ref(result)
    if g_object_is_floating(result.impl).int != 0:
      discard g_object_ref_sink(result.impl)
    g_object_add_toggle_ref(result.impl, toggleNotify, addr(result[]))
    g_object_unref(result.impl)
    assert(g_object_get_qdata(result.impl, Quark) == nil)
    g_object_set_qdata(result.impl, Quark, addr(result[]))
```

The errors occurs in the first line of each function. The same error also occurs using gtk3 code.

**Comment by StefanSalewski** [comment]
Thanks for reporting.

Are you sure that you installed gintro with

```
nimble install gintro@#head
```

And why compile with nimble? All the examples we gave use something like

```
nim c app0.nim
```

But I will test with nimble, should work too, but is one more unnecessary step, nimble just calls the Nim compiler, which then again calls a C compiler.

Will look at it later this evening and tell you.

**Comment by StefanSalewski** [comment]
Ah well, I think I remember, we had that issue maybe 8 months ago. Due to a compiler change, or a GTK change, I can not remember details currently. Please ensure that you are doing a #head install.

When you wonder why #head is now necessary: Version reached 0.9.9 a year ago, and we do not really want to call next version 1.0, as people may then expects a binding free of any bugs. You can find the details in the README.

**Comment by StefanSalewski** [comment]
Was this issue:

https://github.com/StefanSalewski/gintro/issues/207#issuecomment-1282994861

And yes, I agree that 'nimble install gintro@#head' may be not that nice. Actually, I was going to fully delete gintro from github already, but then we had still one or two users, so I let it stay for now. Maybe someday I will relaunch it, maybe with a different name. I hope and assume the issue is solved now for you, but maybe better use just another Nim GUI toolkit, we have more than 20 of them.

**Comment by dexterdy** [comment]
Ah, I did not know I had to install from head. Thank you for that correction. I will look into other UI kits for Nim

**Comment by SolitudeSF** [comment]
> Version reached 0.9.9 a year ago, and we do not really want to call next version 1.0

Why is that stopping you from tagging a new version?

**Comment by StefanSalewski** [comment]
>Why is that stopping you from tagging a new version?

Because tagging as 1.0.0 would indicate that ALL WORKS PERFECT and that there are no more breaking changes. Tagging as 0.9.9.1 or something like that is ugly. The whole tagging was a bad idea of me, whenever creating a new tag, I had to follow a scheme: Edit nimble file, push code, push tags. Maybe the other way, again I can not remember. By doing this sequence wrong I wasted a few version numbers in the past. But well, when we should still have any users, I may relaunch gintro as 0.1.0 again, or maybe with a different name, maybe as sgb (salewskis gobject bindings). But I am not really aware of any users.
